exonum-java-binding/core/rust/src/storage/proof_map_index.rs

```diff
@@ -169,174 +161,44 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofMapInde
     utils::unwrap_exc_or_default(&env, res)
 }
 
-/// Returns Java-proof object.
+/// Returns proof serialized in protobuf format as a Java byte array.
 #[no_mangle]
 pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofMapIndexProxy_nativeGetProof(
     env: JNIEnv,
     _: JObject,
     map_handle: Handle,
     key: jbyteArray,
-) -> jobject {
+) -> jbyteArray {
     let res = panic::catch_unwind(|| {
         let key = convert_to_key(&env, key)?;
         let proof = match *handle::cast_handle::<IndexType>(map_handle) {
             IndexType::SnapshotIndex(ref map) => map.get_proof(key),
             IndexType::ForkIndex(ref map) => map.get_proof(key),
         };
-
-        Ok(convert_to_java_proof(&env, proof)?.into_inner())
+        env.byte_array_from_slice(&proof.to_bytes())
```

Comment:
Why do the other methods wrap in Result::Ok, and these ones have stopped?

Reply:
Because of the last method call returns exactly that type of object we're interested on. The previous version used to return JObject that had to be converted into internal representation.